tests/src/Traits/OgTrait.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Drupal\joinup\Traits;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\Og;
+use Drupal\og\OgMembershipInterface;
+
+/**
+ * Contains helper methods regarding the organic groups.
+ *
+ * @package src\Traits
+ */
+trait OgTrait {
+
+  /**
+   * User with a given og role is already logged in.
+   *
+   * If the user has more than the required roles, he might have permissions
+   * from the rest of the roles that will lead the test to a false positive.
+   * For this reason, we request check for the specific roles passed.
+   *
+   * @param array $roles
+   *   An array of roles to check.
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   A single role, or multiple comma-separated roles in a single string.
+   *
+   * @return bool
+   *   Returns TRUE if the current logged in user has this role (or roles).
+   */
+  protected function loggedInWithOgRole($roles, EntityInterface $entity) {
+    if (!$this->loggedIn() || !$this->user) {
+      return FALSE;
+    }
+    $user = \Drupal::entityTypeManager()->getStorage('user')->loadUnchanged($this->user->uid);
+    $membership = Og::getMembership($user, $entity);
+    if (empty($membership)) {
+      return FALSE;
+    }
```

Comment:
Actually I was wrong, I just had a look at the documentation of `Og::getMembership()`. If you call it without specifying the state, then it will default to `STATE_ACTIVE` and will only return members. This means I have introduced a bug in my latest OG ticket.